server/src/main/java/org/elasticsearch/index/mapper/ParsedDocument.java

```diff
@@ -35,6 +41,56 @@
 
     private Mapping dynamicMappingsUpdate;
 
+    /**
+     * Create a no-op tombstone document
+     * @param index     the index of the document
+     * @param reason    the reason for the no-op
+     */
+    public static ParsedDocument noopTombstone(String index, String reason) {
+        SourceToParse source = new SourceToParse(index, "", new BytesArray("{}"), XContentType.JSON);
+        ParsedDocument doc = tombstone(source, VersionFieldMapper.INSTANCE, SeqNoFieldMapper.INSTANCE);
+        // Store the reason of a noop as a raw string in the _source field
+        final BytesRef byteRef = new BytesRef(reason);
+        doc.rootDoc().add(new StoredField(SourceFieldMapper.NAME, byteRef.bytes, byteRef.offset, byteRef.length));
+        return doc;
+    }
+
+    /**
+     * Create a delete tombstone document
+     * @param index the index of the deleted document
+     * @param id    the id of the deleted document
+     */
+    public static ParsedDocument deleteTombstone(String index, String id) {
+        Objects.requireNonNull(id);
+        SourceToParse source = new SourceToParse(index, id, new BytesArray("{}"), XContentType.JSON);
+        return tombstone(source, IdFieldMapper.NO_FIELDDATA_INSTANCE, VersionFieldMapper.INSTANCE, SeqNoFieldMapper.INSTANCE);
```

Comment:
are we good using the hardcoded false here for the fielddata? I wonder if somehow it needs to be configurable, but I admit I am not familiar with what is needed in this context.